### PR TITLE
Creation and login to users

### PR DESCRIPTION
**Pull request by AleFalcon**
## Summary
* Installed devise-token-auth
* Configuration devise-token-auth
* Created table User and user in this table

## Trello Card
https://trello.com/c/ttwhI9Iz/23-crear-modelo-de-user-implementar-autenticaci%C3%B3n

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -30,6 +30,7 @@ gem 'jbuilder', '~> 2.5'
 # gem 'redis', '~> 4.0'
 # Use ActiveModel has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
+gem 'devise_token_auth'
```
Comment (nicozar95) [inline Gemfile]:
```suggestion
gem 'devise_token_auth', '~> 1.0.0'
```

Comment (nicozar95) [reply Gemfile]: I believe that with this change, we would not need to extend from `extend Devise::Models` in the User

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
```
Comment (nicozar95) [inline app/controllers/application_controller.rb]: Create an `ApiController` that manages all this stuff. So we have an ApiController like:
```
class ApiController < ApplicationController
  include DeviseTokenAuth::Concerns::SetUserByToken
end
```

And a ApplicationController like:
```
class ApplicationController < ActionController::Base
  protect_from_forgery with: :null_session
end
```

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  extend Devise::Models
```
Comment (nicozar95) [inline app/models/user.rb]: Check if by looking the devise version we could delete this line

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  include DeviseTokenAuth::Concerns::User
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (nicozar95) [inline app/models/user.rb]: We will only need: `devise :database_authenticatable, :registerable`

File: config/initializers/devise_token_auth.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+DeviseTokenAuth.setup do |config|
+  # By default the authorization headers will change after each request. The
+  # client is responsible for keeping track of the changing tokens. Change
+  # this to false to prevent the Authorization header from changing after
+  # each request.
+  # config.change_headers_on_each_request = true
```
Comment (nicozar95) [inline config/initializers/devise_token_auth.rb]:
```suggestion
  config.change_headers_on_each_request = false
```

File: Gemfile (hunk 1 above)
Comment (AleFalcon) [reply Gemfile]: it worked

File: app/models/user.rb (hunk 3 above)
Comment (AleFalcon) [reply app/models/user.rb]: Ready

File: config/initializers/devise_token_auth.rb (hunk 5 above)
Comment (AleFalcon) [reply config/initializers/devise_token_auth.rb]: Ready

File: app/models/user.rb (hunk 4 above)
Comment (AleFalcon) [reply app/models/user.rb]: Ready

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (AleFalcon) [reply app/controllers/application_controller.rb]: Ready

**Pull request merged**